oojs/application.js

```diff
@@ -1,13 +1,45 @@
 $(document).ready(function() {
+    
+  var table = new Table();
+
   $('#roller button.add').on('click', function() {
-    console.log("WAT")
-    $('.dice').append('<div class="die">0</div>');
+    table.add();
   });
 
   $('#roller button.roll').on('click', function() {
-    $('.die').each(function(k, die) {
-      var value = Math.floor((Math.random()*6)+1);
-      $(die).text(value);
-    });
+    table.roll();
   });
+
 });
+
+function Table(){
+  this.table = [];
+
+  this.add = function(){
+    var dice = new Dice(this.table.length);
+    dice.show();
+    this.table.push(dice);
+  };
+
+  this.roll = function(){
+    for(var i=0; i<this.table.length; i++){
+      this.table[i].roll();
+    }
+  };
+}
+
+function Dice(id){
```

Comment:
Instead of appending the html and forgetting it, you can make it an attribute of `Dice` so that you can just access it from the JS. With this, you don't need an `id` attribute to find it again.